client/web/web_test.go

```diff
@@ -62,3 +70,59 @@ func TestQnapAuthnURL(t *testing.T) {
 		})
 	}
 }
+
+// TestServeAPI tests the web client api's handling of
+//  1. invalid endpoint errors
+//  2. localapi proxy allowlist
+func TestServeAPI(t *testing.T) {
+	lal := memnet.Listen("local-tailscaled.sock:80")
+	defer lal.Close()
+
+	// Serve dummy localapi. Just returns "success".
+	go func() {
+		localapi := &http.Server{Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintf(w, "success")
+		})}
+		defer localapi.Close()
+		if err := localapi.Serve(lal); err != nil {
+			t.Error(err)
+		}
+	}()
+	s := &Server{lc: &tailscale.LocalClient{Dial: lal.Dial}}
+
+	tests := []struct {
+		name     string
+		reqPath  string
+		wantResp string
+	}{{
+		name:     "invalid_endpoint",
+		reqPath:  "/not-an-endpoint",
+		wantResp: "invalid endpoint",
+	}, {
+		name:     "allowlist_check",
+		reqPath:  "/local/v0/not-allowlisted",
+		wantResp: "/v0/not-allowlisted not allowed from webapi proxy",
+	}, {
+		name:     "localhost_works",
```

Comment:
maybe "logout_works" ?  We're not actually testing hostnames anymore.

Reply:
nice catch, ty!

Reply:
Will make this one `in_localapi_allowlist` and above `not_in_localapi_allowlist`.